### PR TITLE
Fix readme, but for real this time.

### PR DESCRIPTION
**Pull request by tikue**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -251,23 +255,19 @@ the macro automatically chooses `tarpc::util::Never` as the error type.
...
 ```
 
 and the following future-based trait:
 
-```rust
-impl FutureService for HelloServer {
-    type HelloFut = Result<String, Message>;
+```rust,ignore
+trait FutureService for HelloServer {
+    type HelloFut = IntoFutue<String, Message>;
```
Comment (compressed) [inline README.md]: `IntoFuture` I think.

Comment (tikue) [inline README.md]: Done.

File: README.md [hunk 2]
```diff
@@ -41,8 +41,12 @@ tarpc = { git = "https://github.com/google/tarpc" }
 tarpc-plugins = { git = "https://github.com/google/tarpc" }
 ```
 
-## Example
-```rust
+## Example: Sync
+
+tarpc has two APIs: `sync` for blocking requests and `future` for asynchronous
```
Comment (shaladdle) [inline README.md]: nit: How about "`sync` for blocking code and `future` for asynchronous code"?

File: README.md [hunk 3]
```diff
@@ -41,8 +41,12 @@ tarpc = { git = "https://github.com/google/tarpc" }
...
+requests. Here's how to use the sync api.
+
+```rust,no_run
 // required by `FutureClient` (not used directly in this example)
 #![feature(conservative_impl_trait, plugin)]
 #![plugin(tarpc_plugins)]
```
Comment (shaladdle) [inline README.md]: Is `extern crate futures` used in the sync example? What about `tokio_core`?

File: hooks/pre-push [hunk 4]
```diff
@@ -125,7 +125,7 @@ else
               printf "${SUCCESS}\n"
         fi
 
-    run_cargo build
+        run_cargo build
 	run_cargo test
```
Comment (shaladdle) [inline hooks/pre-push]: Nit: Can you fix this to make the indentation consistent? Is the problem that some indents are tabs and some are spaces?

File: README.md (hunk 2 above)
Comment (tikue) [inline README.md]: Done.

File: hooks/pre-push (hunk 4 above)
Comment (tikue) [inline hooks/pre-push]: Fixed hopefully

File: README.md (hunk 3 above)
Comment (tikue) [inline README.md]: Fixed

**Comment by shaladdle** [comment]
Woo!

**Pull request merged**